// components/Article/CreateArticle/index.js
import React, { useEffect, useContext} from 'react'
import {Redirect} from 'react-router-dom'

import ArticleForm from '../ArticleForm'
import {CurrentUserContext} from 'contexts/currentUser'
import firebase from "../../../../firebase";
import useToast from "components/Toaster/useToast";
import {getMonthDay, toDashedLatin} from '../../../utils';
import {ArticlesDataContext} from "contexts/articlesData";

const CreateArticle = () => {
    const [currentUserState] = useContext(CurrentUserContext);
    const toast = useToast();
    const [articlesData, dispatch] = useContext(ArticlesDataContext);

    const onSubmit = (article) => {
        article.slug = toDashedLatin(article.title);
        article.postedOn = getMonthDay();
         firebase.addArticle(article).then(ref => {
             dispatch({type: 'ADD_ARTICLE', payload: [...articlesData.articles, article]});
             toast.add('Article was created!', 'success');
        })
         .catch(error => {
             toast.add(`${error}`, 'danger');
         });
    };

    const initialValues = {
        title: '',
        description: '',
        body: '',
        tagList: '',
        articlePreview: '',
        isPublished: false,
        id: '',
    };

    useEffect(() => {
        if (currentUserState.isLoggedIn === null) {
            return;
        }

        if (currentUserState.isLoggedIn === false) {
            return <Redirect to="/"/>
        }
    },[currentUserState]);

    return (
        <div>
            <ArticleForm
                onSubmit={onSubmit}
                initialValues={initialValues}
            />
        </div>
    )
};

export default CreateArticle;
