// components/Article/EditArticle/index.js
import React, {useState, useEffect, useContext} from 'react'
import {Redirect} from 'react-router-dom'

import ArticleForm from '../ArticleForm'
import {CurrentUserContext} from 'contexts/currentUser'
import firebase from "../../../../firebase";
import withToastProvider from "components/Toaster/withToastProvider";
import useToast from "components/Toaster/useToast";

const EditArticle = ({location, history}) => {
    const [currentUserState] = useContext(CurrentUserContext);
    const [initialValues, setInitialValues] = useState(null)
    const toast = useToast();

    const onSubmit = (newArticle) => {
        const {state: {data}} = location;
        const obj = Object.assign({}, data, newArticle);
        firebase.updateArticle(obj).then(ref => {
            toast.add('Article was updated!', 'success');
        })
            .catch(error => {
                toast.add(`${error}`, 'danger');
            });
    };

    useEffect(() => {
        if (currentUserState.isLoggedIn === null) {
            return;
        }

        if (currentUserState.isLoggedIn === false) {
            return <Redirect to="/"/>
        }
        const {state: {data}} = location;
        if (!data) {
            history.replace('/')
        }

        setInitialValues({
            title: data.title,
            description: data.description,
            body: data.body,
            tagList: data.tagList.join(' '),
            articlePreview: data.articlePreview ? data.articlePreview : '',
            isPublished: !!data.isPublished,
        })
    },[currentUserState]);

    return (
        <div>
            <ArticleForm
                onSubmit={onSubmit}
                initialValues={initialValues}
            />
        </div>
    )
};

export default withToastProvider(EditArticle);
